AlpsSDK/AlpsManager/AlpsManager+Convenience.swift

```diff
@@ -10,41 +10,29 @@ import Foundation
 import Alps
 
 extension AlpsManager {
-    func createMainDevice(device: MobileDevice? = nil, completion: ((MobileDevice?) -> Void)? = nil) {
+    func createMainDevice(device: MobileDevice? = nil, completion: ((Result<MobileDevice?>) -> Void)? = nil) {
         let uiDevice = UIDevice.current
         let location = Location(latitude: 10, longitude: 10, altitude: 10, horizontalAccuracy: 10, verticalAccuracy: 10) // location will be optional in the future
         let mobileDevice = MobileDevice(name: device?.name ?? uiDevice.name,
                                         platform: device?.platform ?? uiDevice.systemName,
                                         deviceToken: device?.deviceToken ?? "",
                                         location: device?.location ?? location)
         mobileDevices.create(item: mobileDevice) { (result) in
-            if case let .success(createdDevice) = result {
-                completion?(createdDevice)
-            } else {
-                completion?(nil)
-            }
+            completion?(result)
```

Comment:
I also think that giving back the result is better than just the object. The developer can know the error and something went wrong.